engine/src/main/java/com/datatorrent/stram/StramLocalCluster.java

```diff
@@ -308,6 +313,24 @@ public StramLocalCluster(LogicalPlan dag) throws IOException, ClassNotFoundExcep
     this.dnmgr = new StreamingContainerManager(dag);
     this.umbilical = new UmbilicalProtocolLocalImpl();
 
+    Method addURL;
+    URLClassLoader loader;
+    try {
+      loader = (URLClassLoader)ClassLoader.getSystemClassLoader();
+      addURL = URLClassLoader.class.getDeclaredMethod("addURL", new Class[] { URL.class });
+      addURL.setAccessible(true);
+    } catch (NoSuchMethodException e) {
+      throw new RuntimeException("No method named addURL found.", e);
+    }
+
+    for (String jarPath : dag.getJarResources().keySet()) {
+      try {
+        addToClasspath(jarPath, loader, addURL);
+      } catch (InvocationTargetException | IllegalAccessException e) {
```

Comment:
StramLocalCluster is mostly used in unit tests that handle Exception. I don't see a big problem with propagating other exception in addition to IOException.